### ISSUE TITLE
I got this error after the new update to 11.8.2

### ISSUE THREAD

**Issue by zifahm** [wbkd/react-flow#3330]
### Describe the Bug

After updating to the latest version from 11.8.1 to 11.8.2

i get this error when I start my app

```
  Background,
  ^^^^^^^^^^
SyntaxError: Named export 'Background' not found. The requested module 'reactflow' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'reactflow';

at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
at ModuleJob.run (node:internal/modules/esm/module_job:190:5)
```
    
   Only workaround right now is to go back to the old version.
   
   Do you guys know how to fix this?

### Steps to Reproduce the Bug or Issue

1. update to version 11.8.2
2. start the app

### Expected behavior

Start without any errors

### Platform

- OS: Linux
- Browser: Chrome
- Version: 11.8.2

**Comment by zifahm** [comment]
using `type: module`

**Comment by zifahm** [comment]
clearly its looking for `.mjs` files and compaining about it being `.js`. Thus it thinks it a commonjs module.

The last update is causing the issue

**Comment by moklick** [comment]
How does your setup look like?

**Comment by zifahm** [comment]
using Remix run with express starter template

**Comment by moklick** [comment]
😵‍💫 I tested with `remix build` and `remix dev` - both things work. `remix-serve build` seems to fail though .. I'll check it. If you have any hints let me know.

**Comment by zifahm** [comment]
The `Express Template` has  `serverModuleFormat` is set to `esm` by the way. And `type: modue` in package.json.

All the other starter templates are in `cjs`

I still get the same error with a fresh install of `express template`

**Comment by zifahm** [comment]
wait should'nt pakckage.jons `moduel` always end in `.mjs` to supprot esm modules?
I think the earlier version you had was correct.
You where supporting both in the last version.

everthing was correct

```
  "module": "dist/esm/index.mjs",
  "types": "dist/esm/index.d.ts",
  "exports": {
    ".": {
      "types": "./dist/esm/index.d.ts",
      "import": "./dist/esm/index.mjs",
      "require": "./dist/umd/index.js"
    }
  ````

**Comment by moklick** [comment]
This doesn't work with React17 :/

**Comment by zifahm** [comment]
:) it's time to do a version bump haha.

reactflow 12 `esm` only!

**Comment by moklick** [comment]
Yep, v12 is React18 only. Then we can safely use mjs - I hate tooling

**Comment by zifahm** [comment]
Yup, this will force the React 17 users to upgrade to 18 as well.

**Comment by moklick** [comment]
Fixed in 11.9.0
